tezos/messages/src/p2p/encoding/block_header.rs

```diff
@@ -92,11 +92,49 @@ pub struct BlockHeader {
     #[get = "pub"]
     #[encoding(list, builtin = "Uint8")]
     protocol_data: Vec<u8>,
+
+    #[get = "pub"]
+    #[serde(skip)]
+    #[builder(default)]
+    #[encoding(hash)]
+    hash: EncodingHash,
+}
+
+#[derive(Clone, Debug)]
+pub struct EncodingHash(pub Option<Vec<u8>>);
```

Comment:
@akoptelov 
I would be more specific here:
`EncodingHash` -> `EncodingBlockHash`

Reply:
It is general enough, probably it will be moved to some common place to be used in other messages.